eng/common/TestResources/New-TestResources.ps1

```diff
@@ -313,25 +311,53 @@ foreach ($templateFile in $templateFiles) {
         }
     }
 
-    foreach ($key in $deploymentOutputs.Keys)
+    if ($File)
     {
-        $value = $deploymentOutputs[$key]
-        
-        if ($CI) {
-            # Treat all ARM template output variables as secrets since "SecureString" variables do not set values.
-            # In order to mask secrets but set environment variables for any given ARM template, we set variables twice as shown below.
-            Write-Host "Setting variable '$key': ***"
-            Write-Host "##vso[task.setvariable variable=_$key;issecret=true;]$($value)"
-            Write-Host "##vso[task.setvariable variable=$key;]$($value)"
-        } else {
-            Write-Host ($shellExportFormat -f $key, $value)
+        if (!($IsWindows))
+        {
+            Write-Host "File option is supported only on Windows"
         }
+
+        $outputPath = Join-Path $env:USERPROFILE ".Azure" "TestEnvironments"
+        $outputFile = Join-Path $outputPath $serviceName
+
+        $environmentText = $deploymentOutputs | ConvertTo-Json;
+        $bytes = ([System.Text.Encoding]::UTF8).GetBytes($environmentText)
+        $protectedBytes = [Security.Cryptography.ProtectedData]::Protect($bytes, $null, [Security.Cryptography.DataProtectionScope]::LocalMachine)
+
+        New-Item -Type Directory $outputPath -Force | Out-Null
+        Set-Content $outputFile -Value $protectedBytes -AsByteStream -Force
+
+        Write-Host "Test environment settings`n $environmentText`nstored into encrypted $outputFile"
     }
+    else
+    {
+        
+        if (!$CI) {
```

Comment:
Rather than nesting the whole block and shifting everything over, change line 333 to `elif (!$CI)`.

Reply:
There is more non-CI code in this branch.